src/Kestrel.Core/Internal/Infrastructure/LoggingThreadPool.cs

```diff
@@ -50,12 +50,7 @@ public override void UnsafeRun(WaitCallback action, object state)
             System.Threading.ThreadPool.QueueUserWorkItem(action, state);
         }
 
-        public override void Schedule(Action action)
-        {
-            Run(action);
-        }
-
-        public override void Schedule(Action<object> action, object state)
+        public override void Schedule<T>(Action<T> action, T state)
         {
```

Comment:
/cc @halter73 this allocates per schedule now.

Reply:
Have we measured how this affects our benchmarks?

Reply:
Nope